In [1]:
#Dependencies and Setup
import pandas as pd
import numpy as np

In [2]:
#File to Load 
    file_to_load = "Resources/purchase_data.csv"
    
#Read Purchasing File and store into Pandas data frame
    purchase_data = pd.read_csv(file_to_load)
   

In [3]:
#Use Pandas to read data
purchase_data_df = pd.read_csv(file_to_load)
purchase_data_df.head()



Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [4]:
#Display a statistical overview of the purchase data.
purchase_data.describe()

Purchase ID         Age     Item ID       Price
count   780.000000  780.000000  780.000000  780.000000
mean    389.500000   22.714103   92.114103    3.050987
std     225.310896    6.659444   52.775943    1.169549
min       0.000000    7.000000    0.000000    1.000000
25%     194.750000   20.000000   48.000000    1.980000
50%     389.500000   22.000000   93.000000    3.150000
75%     584.250000   25.000000  139.000000    4.080000
max     779.000000   45.000000  183.000000    4.990000

# Player Count

In [5]:
players_df = purchase_data_df.groupby("SN")["SN"].nunique()
players_df.count()

576

In [6]:
players_count = len(purchase_data_df["SN"].unique())

In [7]:
#Display the total number of players
summary_table = pd.DataFrame({"Total players": [players_count]})
                              
summary_table

Total players
0            576

# Purchasing Analysis (Total)

#### - Run basic calculations to obtain number of unique items, average price, etc.
#### - Create a summary data frame to hold the results
#### - optional: give the displayed data cleaner formatting
#### - Display the summary data frame

In [34]:
#Number of unique items.
items_df = len(purchase_data_df["Item ID"].unique())

#Avarage price calculation
average_price_df = purchase_data_df["Price"].mean()
average_price_df

#Total number of purchases.
total_purchases_df = len(purchase_data_df)
total_purchases_df


#Total Revenue.
total_revenue_df = purchase_data_df["Price"].sum()
total_revenue_df.round(2)



#Inserting result into a DAtaFrame
purchasing_analysis_df = pd.DataFrame({"Number of Unique Items":[items_df],
                                             "Average Price":[average_price_df],
                                            "Number of Purchases":[total_purchases_df],
                                             "Total Revenue":[total_revenue_df]})

purchasing_analysis_df = purchasing_analysis_df.round(2)
purchasing_analysis_df["Average Price"] = purchasing_analysis_df["Average Price"].map("${:,.2f}".format)
purchasing_analysis_df["Total Revenue"] = purchasing_analysis_df["Total Revenue"].map("${:,.2f}".format)
purchasing_analysis_df = purchasing_analysis_df.loc[:, ["Number of Unique Items", "Average Price", "Number of Purchases", "Total Revenue"]]


purchasing_analysis_df

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     183         $3.05                  780     $2,379.77

# Gender Demographics

#### -Percentage and Count of Male Players
#### -Percentage and Count of Female Players
#### -Percentage and Count of Other / Non-Disclosed

In [9]:
#Gender Count
gender_count_df = purchase_data_df.groupby("Gender")["SN"].nunique()
gender_count_df.head()

#Gender Percentage
gender_percentage_df = (gender_count_df/576)*100
gender_percentage_df.round(2)


#Gender demographics DataFrame.
gender_demographics_df = pd.DataFrame({"Total Count": gender_count_df,
                                       "Percentage of Players":gender_percentage_df})

gender_demographics_df = gender_demographics_df.round(2)

gender_demographics_df

Total Count  Percentage of Players
Gender                                                   
Female                          81                  14.06
Male                           484                  84.03
Other / Non-Disclosed           11                   1.91

# Purchasing Analysis (Gender)

#### - Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. by gender
#### -Create a summary data frame to hold the results
#### -Optional: give the displayed data cleaner formatting
#### -Display the summary data frame

In [36]:
#Purchase count
gender_purchase_count_df= purchase_data_df.groupby("Gender")["SN"].count()
gender_purchase_count_df

#Average price of items by gender.
gender_average_df = purchase_data_df.groupby("Gender")["Price"].mean()
gender_average_df.round(2)

#Total Purchase value by gender.
gender_total_purchases_df = purchase_data_df.groupby("Gender")["Price"].sum()
gender_total_purchases_df

#Average Total purchase per person
gender_total_revenue_df = gender_total_purchases_df/gender_count_df
gender_total_revenue_df.round(2)


#Purchasing analysis DataFrame by gender.
gender_analysis_df = pd.DataFrame({"Purchase Count":gender_purchase_count_df,
                                             "Average Purchase Price":gender_average_df,
                                             "Total Purchase Value":gender_total_purchases_df,
                                             "Avg Total Purchase per Person":gender_total_revenue_df
                                          })

gender_analysis_df = gender_analysis_df.round(2)
gender_analysis_df["Average Purchase Price"] = gender_analysis_df["Average Purchase Price"].map("${:,.2f}".format)
gender_analysis_df["Total Purchase Value"] = gender_analysis_df["Total Purchase Value"].map("${:,.2f}".format)
gender_analysis_df["Avg Total Purchase per Person"] = gender_analysis_df["Avg Total Purchase per Person"].map("${:,.2f}".format)


gender_analysis_df




Purchase Count Average Purchase Price  \
Gender                                                         
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg Total Purchase per Person  
Gender                                                                    
Female                             $361.94                         $4.47  
Male                             $1,967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

In [11]:
#Create Bins  <10, 10-14, 15-19, 20-24, 25-29, 30-34, 35-39 >39.
bins = [0,9.99,14.99,19.99,24.99,29.99,34.99,39.99, 9999]
age_ranges = ["<10", "10-14","15-19", "20-24", "25-29", "30-34", "35-39", ">40"]

In [12]:
#Cut purchase data and place the ages into bins
pd.cut(purchase_data_df["Age"], bins, labels=age_ranges)

0      20-24
1        >40
2      20-24
3      20-24
4      20-24
       ...  
775    20-24
776    20-24
777    20-24
778      <10
779    20-24
Name: Age, Length: 780, dtype: category
Categories (8, object): [<10 < 10-14 < 15-19 < 20-24 < 25-29 < 30-34 < 35-39 < >40]

In [13]:
#Adding the column Age Range to the Data Frame
purchase_data_df["Age Range"] = pd.cut(purchase_data_df["Age"], bins, labels= age_ranges)
purchase_data_df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Range  
0  Extraction, Quickblade Of Trembling Hands   3.53     20-24  
1                          Frenzied Scimitar   1.56       >40  
2                               Final Critic   4.88     20-24  
3                                Blindscythe   3.27     20-24  
4                                       Fury   1.44     20-24

In [14]:
#Creating a group based of the bins
by_age_df = purchase_data_df.groupby("Age Range")
by_age_df.max()

Purchase ID             SN  Age                 Gender  Item ID  \
Age Range                                                                    
<10                778     Tyeurith29    9                   Male      179   
10-14              774     Zhisrisu83   14                   Male      180   
15-19              771  Yoishirrala98   19  Other / Non-Disclosed      183   
20-24              779      Zontibe81   24  Other / Non-Disclosed      183   
25-29              772        Yasur35   29  Other / Non-Disclosed      183   
30-34              770     Yarolwen77   34                   Male      178   
35-39              768     Tyaerith73   39  Other / Non-Disclosed      179   
>40                761      Salilis27   45                   Male      173   

                                 Item Name  Price  
Age Range                                          
<10        Wolf, Promise of the Moonwalker   4.93  
10-14                                 Wolf   4.94  
15-19                   Yearning Mageblade   4.91  
20-24                   Yearning Mageblade   4.99  
25-29                         Worldbreaker   4.94  
30-34                   Yearning Mageblade   4.93  
35-39      Wolf, Promise of the Moonwalker   4.91  
>40                     Victor Iron Spikes   4.93

# Age Demographics

#### - Establish bins for ages
#### - Categorize the existing players using the age bins. Hint: use pd.cut()
#### - Calculate the numbers and percentages by age group
#### - Create a summary data frame to hold the results
#### - Optional: round the percentage column to two decimal points
#### - Display Age Demographics Table

In [15]:
#Total count by Age Range
age_group_count_df = purchase_data_df.groupby("Age Range")["SN"].nunique()
age_group_count_df.round(2)

#Gender Percentage
count_percentage_df = (age_group_count_df/576)*100

#Insert result into DataFrame
age_demo_df = pd.DataFrame({"Total Count":age_group_count_df,
                            "Percentage of Players":count_percentage_df,
                            
})
age_demo_df=age_demo_df.round(2)
age_demo_df

Total Count  Percentage of Players
Age Range                                    
<10                 17                   2.95
10-14               22                   3.82
15-19              107                  18.58
20-24              258                  44.79
25-29               77                  13.37
30-34               52                   9.03
35-39               31                   5.38
>40                 12                   2.08

# Purchasing Analysis (Age)

#### - Bin the purchase_data data frame by age
#### - Run basic calculations to obtain purchase count, avg. purchase price, avg. purchase total per person etc. in the table below
#### - Create a summary data frame to hold the results
#### - Optional: give the displayed data cleaner formatting
#### - Display the summary data frame

In [37]:
#Purchase count by age range.
age_group_count_df = purchase_data_df.groupby("Age Range")["SN"].count()
age_group_count_df


#Purchase count by age range.
age_group_count_df = purchase_data_df.groupby("Age Range")["SN"].count()
age_group_count_df


#Average purchase price by age range.
age_group_average_df = purchase_data_df.groupby("Age Range")["Price"].mean()
age_group_average_df.round(2)


#Total purchase value by age range.
age_group_total_df = purchase_data_df.groupby("Age Range")["Price"].sum()
age_group_total_df

#Avg Total Purchase per Person
normalized_age_total_df = age_group_total_df/576
normalized_age_total_df.round(2)

# Convert to DataFrame.
age_range_df = pd.DataFrame({"Purchase Count":age_group_count_df,
                            "Average Purchase Price":age_group_average_df,
                            "Total Purchase Value": age_group_total_df,
                            "Avg Total Purchase per Person": normalized_age_total_df
})

age_range_df = age_range_df.round(2)
age_range_df["Average Purchase Price"] = age_range_df["Average Purchase Price"].map("${:,.2f}".format)
age_range_df["Total Purchase Value"] = age_range_df["Total Purchase Value"].map("${:,.2f}".format)
age_range_df["Avg Total Purchase per Person"] = age_range_df["Avg Total Purchase per Person"].map("${:,.2f}".format)
age_range_df

Purchase Count Average Purchase Price Total Purchase Value  \
Age Range                                                               
<10                    23                  $3.35               $77.13   
10-14                  28                  $2.96               $82.78   
15-19                 136                  $3.04              $412.89   
20-24                 365                  $3.05            $1,114.06   
25-29                 101                  $2.90              $293.00   
30-34                  73                  $2.93              $214.00   
35-39                  41                  $3.60              $147.67   
>40                    13                  $2.94               $38.24   

          Avg Total Purchase per Person  
Age Range                                
<10                               $0.13  
10-14                             $0.14  
15-19                             $0.72  
20-24                             $1.93  
25-29                             $0.51  
30-34                             $0.37  
35-39                             $0.26  
>40                               $0.07

# Top Spenders

#### - Run basic calculations to obtain the results in the table below
#### - Create a summary data frame to hold the results
#### - Sort the total purchase value column in descending order
#### - Optional: give the displayed data cleaner formatting
#### - Display a preview of the summary data frame

In [38]:
#Purchase count by sn
sn_group_count_df = purchase_data_df.groupby("SN")["Item ID"].count()
sn_group_count_df

#Average purchase price by sn.
sn_group_average_df = purchase_data_df.groupby("SN")["Price"].mean()
sn_group_average_df.round(2)

#Total purchase value by sn.
sn_group_total_df = purchase_data_df.groupby("SN")["Price"].sum()
sn_group_total_df# Convert to DataFrame.
sn_range_df = pd.DataFrame({"Purchase Count":sn_group_count_df,
                            "Average Purchase Price":sn_group_average_df,
                            "Total Purchase Value": sn_group_total_df,
                            
})
sn_range_df = sn_range_df.round(2)
sn_range_df["Average Purchase Price"] = sn_range_df["Average Purchase Price"].map("${:,.2f}".format)
sn_range_df["Total Purchase Value"] = sn_range_df["Total Purchase Value"].map("${:,.2f}".format)

sn_range_df

Purchase Count Average Purchase Price Total Purchase Value
SN                                                                       
Adairialis76                1                  $2.28                $2.28
Adastirin33                 1                  $4.48                $4.48
Aeda94                      1                  $4.91                $4.91
Aela59                      1                  $4.32                $4.32
Aelaria33                   1                  $1.79                $1.79
...                       ...                    ...                  ...
Yathecal82                  3                  $2.07                $6.22
Yathedeu43                  2                  $3.01                $6.02
Yoishirrala98               1                  $4.58                $4.58
Zhisrisu83                  2                  $3.94                $7.89
Zontibe81                   3                  $2.68                $8.03

[576 rows x 3 columns]

In [32]:
#Total items purchases analysis.

items_purchase_count_df = purchase_data_df.groupby(["Item ID", "Item Name"]).count()["Price"]
items_average_price_df = purchase_data_df.groupby(["Item ID", "Item Name"]).mean()["Price"]
items_value_total_df = purchase_data_df.groupby(["Item ID", "Item Name"]).sum()["Price"]

items_purchased_df = pd.DataFrame({"Purchase Count":items_purchase_count_df,
                                   "Item Price":items_average_price_df,
                                   "Total Purchase Value":items_value_total_df,})

items_purchased_df = items_purchased_df.round(2)
items_purchased_df["Item Price"] = items_purchased_df["Item Price"].map("${:,.2f}".format)
items_purchased_df["Total Purchase Value"] = items_purchased_df["Total Purchase Value"].map("${:,.2f}".format)

items_purchased_df.head()



Purchase Count Item Price Total Purchase Value
Item ID Item Name                                                         
0       Splinter                         4      $1.28                $5.12
1       Crucifer                         3      $3.26                $9.78
2       Verdict                          6      $2.48               $14.88
3       Phantomlight                     6      $2.49               $14.94
4       Bloodlord's Fetish               5      $1.70                $8.50

# Most Popular Items

#### - Retrieve the Item ID, Item Name, and Item Price columns
#### - Group by Item ID and Item Name. Perform calculations to obtain purchase count, item price, and total purchase value
#### - Create a summary data frame to hold the results
#### - Sort the purchase count column in descending order
#### - Optional: give the displayed data cleaner formatting
#### - Display a preview of the summary data frame

In [33]:
#Sort table to show the five the most popular items.
most_popular_items_df = items_purchased_df.sort_values("Purchase Count", ascending=False)
most_popular_items_df

Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
108     Extraction, Quickblade Of Trembling Hands                  9   
82      Nirvana                                                    9   
19      Pursuit, Cudgel of Necromancy                              8   
...                                                              ...   
104     Gladiator's Glaive                                         1   
23      Crucifer                                                   1   
180     Stormcaller                                                1   
91      Celeste                                                    1   
134     Undead Crusader                                            1   

                                                     Item Price  \
Item ID Item Name                                                 
178     Oathbreaker, Last Hope of the Breaking Storm      $4.23   
145     Fiery Glass Crusader                              $4.58   
108     Extraction, Quickblade Of Trembling Hands         $3.53   
82      Nirvana                                           $4.90   
19      Pursuit, Cudgel of Necromancy                     $1.02   
...                                                         ...   
104     Gladiator's Glaive                                $1.93   
23      Crucifer                                          $1.99   
180     Stormcaller                                       $3.36   
91      Celeste                                           $4.17   
134     Undead Crusader                                   $4.50   

                                                     Total Purchase Value  
Item ID Item Name                                                          
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
108     Extraction, Quickblade Of Trembling Hands                  $31.77  
82      Nirvana                                                    $44.10  
19      Pursuit, Cudgel of Necromancy                               $8.16  
...                                                                   ...  
104     Gladiator's Glaive                                          $1.93  
23      Crucifer                                                    $1.99  
180     Stormcaller                                                 $3.36  
91      Celeste                                                     $4.17  
134     Undead Crusader                                             $4.50  

[183 rows x 3 columns]

In [31]:
#Sort table to show the five the most profitable items.

most_profitable_items_df = items_purchased_df.sort_values("Total Purchase Value", ascending=False)
most_profitable_items_df



Purchase Count  \
Item ID Item Name                                                      
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
92      Final Critic                                               8   
103     Singed Scalpel                                             8   
...                                                              ...   
125     Whistling Mithril Warblade                                 2   
126     Exiled Mithril Longsword                                   1   
23      Crucifer                                                   1   
104     Gladiator's Glaive                                         1   
42      The Decapitator                                            1   

                                                      Item Price  \
Item ID Item Name                                                  
178     Oathbreaker, Last Hope of the Breaking Storm        4.23   
82      Nirvana                                             4.90   
145     Fiery Glass Crusader                                4.58   
92      Final Critic                                        4.88   
103     Singed Scalpel                                      4.35   
...                                                          ...   
125     Whistling Mithril Warblade                          1.00   
126     Exiled Mithril Longsword                            2.00   
23      Crucifer                                            1.99   
104     Gladiator's Glaive                                  1.93   
42      The Decapitator                                     1.75   

                                                      Total Purchase Value  
Item ID Item Name                                                           
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
82      Nirvana                                                      44.10  
145     Fiery Glass Crusader                                         41.22  
92      Final Critic                                                 39.04  
103     Singed Scalpel                                               34.80  
...                                                                    ...  
125     Whistling Mithril Warblade                                    2.00  
126     Exiled Mithril Longsword                                      2.00  
23      Crucifer                                                      1.99  
104     Gladiator's Glaive                                            1.93  
42      The Decapitator                                               1.75  

[183 rows x 3 columns]

# Conclusion

## 1. First conclusion is that Males - counting 484 in total represents 84.03% of total players compared with Females - that is counting only 81 individuals, with a percentage of 14.61%.2. After sorting by Age Range, we can notice that  most of the players are included in the range of 20-25 (258 players representing 44.79%) and 15-20 (106 players). 3. The average price is about 3 dollars per purchase with the top spenders (age range of 35-39) paying up to 3,60 dollars for their purchases.The most expensive games are bought by the players in the range of 35-39. 